benchmarks/dynamo/pr_time_benchmarks/benchmarks/update_hint_benchmark.py

```diff
@@ -34,12 +34,15 @@ def f(a, b):
                 torch._check(x <= self.N)
             return a.split(xs)
 
-        f(self.input, self.splits)
+        for i in range(1000):
+            f(self.input, self.splits)
```

Comment:
If you want to be in the business of measuring the instruction count of compiled codes, it seems like it would be better to do a measurement here that *excludes* compilation time from the perf measurement. The most typical way to do this would be to do the warmup (forcing compilation) and then do the perf measurement in steady-state. Otherwise, your overall instruction count number is going to bounce up and down, mostly due to changes in compile time, not so useful for finding runtime regressions unless you take care to subtract out the compile time.

Reply:
the instruction count metric, measure the total instruction count for the work function, this could be something that does not even do compilation . (people can add micro benchmarks). 

I see your point about adding another metric which is " instruction count of compiled codes" but that would be a different diff. 

for the time being i can exclude the enable_instruction_count() for the update hint benchmark since what we care about for that one is compile time instruction count. 

I was using that as basic benchmark example, maybe i will add a dummy benchmark were i enable everything on it as example.